optimum/neuron/modeling_decoder.py

```diff
@@ -0,0 +1,208 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for text-generation model architectures on neuron devices."""
+
+import logging
+import os
+import shutil
+from pathlib import Path
+from tempfile import TemporaryDirectory
+from typing import TYPE_CHECKING, Optional, Tuple, Union
+
+import torch
+from transformers import GenerationConfig
+
+from ..exporters.neuron.model_configs import *  # noqa: F403
+from ..exporters.tasks import TasksManager
+from ..modeling_base import OptimizedModel
+from .utils import is_transformers_neuronx_available
+
+
+if is_transformers_neuronx_available():
+    from transformers_neuronx.module import PretrainedModel as NeuronxPretrainedModel
+    from transformers_neuronx.module import save_pretrained_split
+
+
+if TYPE_CHECKING:
+    from transformers import PretrainedConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_exporter(config, task):
+    return TasksManager.get_exporter_config_constructor(model_type=config.model_type, exporter="neuron", task=task)()
+
+
+class NeuronDecoderModel(OptimizedModel):
```

Comment:
Why dont we inherint from `NeuronBaseModel`?

Reply:
Again, this is because NeuronBaseModel is too specific: I looked carefully and there isn't a single method or attribute I could reuse here.